### ISSUE TITLE
[lite]Compiler build warnings 

### ISSUE THREAD

**Issue by samveen** [tensorflow/tensorflow#60261]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

2.12.0

### Custom Code

No

### OS Platform and Distribution

Linux Raspberrypi OS 32-bit (bullseye)  - (Linux 6.1.19+ \# 1637 Tue Mar 14 11:01:56 GMT 2023 armv6l GNU/Linux)

### Mobile device

Raspberry Pi Zero W

### Python version

3.9.2

### Bazel version

CMake build used

### GCC/Compiler version

GNU c++ (Raspbian 10.2.1-6+rpi1) 10.2.1 20210110

### CUDA/cuDNN version

_N.A._

### GPU model and memory

_N.A._

### Current Behaviour?

I get the following warning while building tensorflow-lite with cmake on a Raspberry Pi Zero W running Raspberrypi OS lite 32-bit (Debian bullseye).
```
[ 54%] Building CXX object CMakeFiles/tensorflow-lite.dir/kernels/conv3d.cc.o
/usr/bin/c++ -DCPUINFO_SUPPORTED_PLATFORM=1 -DEIGEN_MPL2_ONLY -DNOMINMAX=1 -DPTHREADPOOL_NO_DEPRECATED_API=1 -I/home/samveen/tensorflow -I/home/samveen/tensorflow/build/gemmlowp -I/home/samveen/tensorflow/build/eigen -I/home/samveen/tensorflow/build/neon2sse -I/home/samveen/tensorflow/build/abseil-cpp -I/home/samveen/tensorflow/build/farmhash/src -I/home/samveen/tensorflow/build/flatbuffers/include -I/home/samveen/tensorflow/build/ruy -I/home/samveen/tensorflow/build/cpuinfo/include -I/home/samveen/tensorflow/build/pthreadpool-source/include -march=armv6 -mfpu=vfp -mfloat-abi=hard -funsafe-math-optimizations -O3 -DNDEBUG -fPIC -DEIGEN_NEON_GEBP_NR=4 -DTFL_STATIC_LIBRARY_BUILD -pthread -std=gnu++17 -o CMakeFiles/tensorflow-lite.dir/kernels/conv3d.cc.o -c /home/samveen/tensorflow/tensorflow/lite/kernels/conv3d.cc
In file included from /home/samveen/tensorflow/build/eigen/Eigen/Core:266,
                 from /home/samveen/tensorflow/third_party/eigen3/Eigen/Core:1,
                 from /home/samveen/tensorflow/tensorflow/lite/kernels/internal/optimized/optimized_ops.h:39,
                 from /home/samveen/tensorflow/tensorflow/lite/kernels/conv3d.cc:25:
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/functors/UnaryFunctors.h: In instantiation of ‘Eigen::internal::scalar_unary_pow_op<Scalar, ExponentScalar, false, false, false, false>::scalar_unary_pow_op(const ExponentScalar&) [with Scalar = float; ExponentScalar = double]’:
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/../plugins/ArrayCwiseUnaryOps.h:714:66:   required from ‘Eigen::ArrayBase<Derived>::UnaryPowReturnType<ScalarExponent> Eigen::ArrayBase<Derived>::pow(const ScalarExponent&) const [with ScalarExponent = double; Derived = Eigen::ArrayWrapper<Eigen::Map<Eigen::Matrix<float, -1, -1>, 0, Eigen::Stride<0, 0> > >; Eigen::ArrayBase<Derived>::UnaryPowReturnType<ScalarExponent> = Eigen::CwiseUnaryOp<Eigen::internal::scalar_unary_pow_op<float, double, false, false, false, false>, const Eigen::ArrayWrapper<Eigen::Map<Eigen::Matrix<float, -1, -1>, 0, Eigen::Stride<0, 0> > > >]’
/home/samveen/tensorflow/tensorflow/lite/kernels/internal/optimized/optimized_ops.h:3327:78:   required from here
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/functors/UnaryFunctors.h:1218:27: warning: ‘std::enable_if_t<(! IsExactlyRepresentable), void> Eigen::internal::scalar_unary_pow_op<Scalar, ExponentScalar, false, false, false, false>::check_is_representable() const [with bool IsExactlyRepresentable = false; Scalar = float; ExponentScalar = double; std::enable_if_t<(! IsExactlyRepresentable), void> = void]’ is deprecated [-Wdeprecated-declarations]
 1218 |     check_is_representable();
      |     ~~~~~~~~~~~~~~~~~~~~~~^~
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/functors/UnaryFunctors.h:1214:68: note: declared here
 1214 |   EIGEN_DEPRECATED std::enable_if_t<!IsExactlyRepresentable, void> check_is_representable() const {}
      |                                                                    ^~~~~~~~~~~~~~~~~~~~~~
```

### Standalone code to reproduce the issue
Standard build on a Pi Zero W running Raspberry Pi OS lite 32-bit (Debian bullseye)
```shell
mkdir build && cd build
ARMCC_FLAGS="-march=armv6 -mfpu=vfp -mfloat-abi=hard -funsafe-math-optimizations"
cmake -DCMAKE_C_FLAGS="${ARMCC_FLAGS}" -DCMAKE_CXX_FLAGS="${ARMCC_FLAGS}"  \
        -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_PROCESSOR=armv6 \
        -DTFLITE_ENABLE_XNNPACK=OFF ../tensorflow/lite/
cmake --build .
```

### Relevant log output

```shell
[ 54%] Building CXX object CMakeFiles/tensorflow-lite.dir/kernels/conv3d.cc.o
/usr/bin/c++ -DCPUINFO_SUPPORTED_PLATFORM=1 -DEIGEN_MPL2_ONLY -DNOMINMAX=1 -DPTHREADPOOL_NO_DEPRECATED_API=1 -I/home/samveen/tensorflow -I/home/samveen/tensorflow/build/gemmlowp -I/home/samveen/tensorflow/build/eigen -I/home/samveen/tensorflow/build/neon2sse -I/home/samveen/tensorflow/build/abseil-cpp -I/home/samveen/tensorflow/build/farmhash/src -I/home/samveen/tensorflow/build/flatbuffers/include -I/home/samveen/tensorflow/build/ruy -I/home/samveen/tensorflow/build/cpuinfo/include -I/home/samveen/tensorflow/build/pthreadpool-source/include -march=armv6 -mfpu=vfp -mfloat-abi=hard -funsafe-math-optimizations -O3 -DNDEBUG -fPIC -DEIGEN_NEON_GEBP_NR=4 -DTFL_STATIC_LIBRARY_BUILD -pthread -std=gnu++17 -o CMakeFiles/tensorflow-lite.dir/kernels/conv3d.cc.o -c /home/samveen/tensorflow/tensorflow/lite/kernels/conv3d.cc
In file included from /home/samveen/tensorflow/build/eigen/Eigen/Core:266,
                 from /home/samveen/tensorflow/third_party/eigen3/Eigen/Core:1,
                 from /home/samveen/tensorflow/tensorflow/lite/kernels/internal/optimized/optimized_ops.h:39,
                 from /home/samveen/tensorflow/tensorflow/lite/kernels/conv3d.cc:25:
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/functors/UnaryFunctors.h: In instantiation of ‘Eigen::internal::scalar_unary_pow_op<Scalar, ExponentScalar, false, false, false, false>::scalar_unary_pow_op(const ExponentScalar&) [with Scalar = float; ExponentScalar = double]’:
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/../plugins/ArrayCwiseUnaryOps.h:714:66:   required from ‘Eigen::ArrayBase<Derived>::UnaryPowReturnType<ScalarExponent> Eigen::ArrayBase<Derived>::pow(const ScalarExponent&) const [with ScalarExponent = double; Derived = Eigen::ArrayWrapper<Eigen::Map<Eigen::Matrix<float, -1, -1>, 0, Eigen::Stride<0, 0> > >; Eigen::ArrayBase<Derived>::UnaryPowReturnType<ScalarExponent> = Eigen::CwiseUnaryOp<Eigen::internal::scalar_unary_pow_op<float, double, false, false, false, false>, const Eigen::ArrayWrapper<Eigen::Map<Eigen::Matrix<float, -1, -1>, 0, Eigen::Stride<0, 0> > > >]’
/home/samveen/tensorflow/tensorflow/lite/kernels/internal/optimized/optimized_ops.h:3327:78:   required from here
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/functors/UnaryFunctors.h:1218:27: warning: ‘std::enable_if_t<(! IsExactlyRepresentable), void> Eigen::internal::scalar_unary_pow_op<Scalar, ExponentScalar, false, false, false, false>::check_is_representable() const [with bool IsExactlyRepresentable = false; Scalar = float; ExponentScalar = double; std::enable_if_t<(! IsExactlyRepresentable), void> = void]’ is deprecated [-Wdeprecated-declarations]
 1218 |     check_is_representable();
      |     ~~~~~~~~~~~~~~~~~~~~~~^~
/home/samveen/tensorflow/build/eigen/Eigen/src/Core/functors/UnaryFunctors.h:1214:68: note: declared here
 1214 |   EIGEN_DEPRECATED std::enable_if_t<!IsExactlyRepresentable, void> check_is_representable() const {}
      |                                                                    ^~~~~~~~~~~~~~~~~~~~~~
```
</details>

**Comment by bhagirath20** [comment]
According to the log output, there appears to be a warning relating to the Eigen library in the TensorFlow source. The warning notice specifically states that a function called check_is_representable() is deprecated.

This warning is most likely generated by the Eigen library being used with a later version of the compiler, which causes the warning.

While this warning should have no effect on TensorFlow's operation, it's always a good idea to avoid utilizing deprecated functions in your code. One method is to upgrade to a more recent version of the library that does not use deprecated functions.

You can also attempt silencing the warning with a compiler flag, although this is generally not recommended because it may hide the warning.

**Comment by pkgoogle** [comment]
Hi @samveen did the above response answer your question?

**Comment by samveen** [comment]
@pkgoogle No, that did not help at all. 

I have made no changes to the codebase, and this warning occurs. The comment is suggesting that I modify the tensorflow build system for the lite build, which defeats the purpose of reporting this issue.

**Comment by pkgoogle** [comment]
Hi @samveen, Are you actually able to compile despite the warning? Generally speaking, warnings will take a long time to get to as they are lower priority. Which README section are you referring to? Thanks for any additional information you can provide.

**Comment by samveen** [comment]
@pkgoogle 

 Please look at the table listed here: 
https://github.com/tensorflow/tensorflow#official-builds

As for the build, if you give me a revision or git tag, I'll run a build for it and give you the results (though it will be slow, as it'll be a native build)

**Comment by pkgoogle** [comment]
Hi @samveen, can you try with nightly? It seems the official build is blocked by a malformed docker issue rather than the actual build.

**Comment by samveen** [comment]
OK. I will run a native 32-bit build for tf lite on my raspberry Pi Zero W and let you know the result. (The build will take quite a bit or time to complete).

**Comment by samveen** [comment]
Progress report:
- Instructions source https://www.tensorflow.org/lite/guide/build_cmake
- Source details:
    ```
    samveen@facez:~/tensorflow_src $ git remote -v
    origin  https://github.com/tensorflow/tensorflow.git (fetch)
    origin  https://github.com/tensorflow/tensorflow.git (push)
    samveen@facez:~/tensorflow_src $ git describe --tags
    v2.12.1
    ```
- CMake command and options:
    ```
    cmake ../tensorflow_src/tensorflow/lite
    ```
- Build command:
    ```
    ~/tflite_build $ cmake --build . -j 1  -t label_image
    ```
- End result: Error with build due to missing options for libatomic in `examples/label_image/CMakeLists.txt`

(snipped)
```
/usr/bin/ld: eigen_support.cc:(.text._ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE[_ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE]+0x778): undefined reference to `__atomic_load_8'
/usr/bin/ld: eigen_support.cc:(.text._ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE[_ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE]+0x7fc): undefined reference to `__atomic_compare_exchange_8'
/usr/bin/ld: eigen_support.cc:(.text._ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE[_ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE]+0x860): undefined reference to `__atomic_load_8'
/usr/bin/ld: eigen_support.cc:(.text._ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE[_ZN14EigenForTFLite15ThreadPoolTemplINS_20StlThreadEnvironmentEE11WaitForWorkEPNS_10EventCount6WaiterEPNS1_4TaskE]+0x8c4): undefined reference to `__atomic_load_8'
collect2: error: ld returned 1 exit status
gmake[3]: *** [examples/label_image/CMakeFiles/label_image.dir/build.make:370: examples/label_image/label_image] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:6608: examples/label_image/CMakeFiles/label_image.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:6615: examples/label_image/CMakeFiles/label_image.dir/rule] Error 2
```

I am modifying the CMake config and restarting the process.

**Comment by samveen** [comment]
I've managed to get target `benchmark_model` build.  However, using the examples as listed in https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/benchmark using the model files from https://storage.googleapis.com/download.tensorflow.org/models/inception5h.zip gives me the following output:

```
samveen@piz:~/tflite_build $ ./tools/benchmark/benchmark_model --graph=../tf/tensorflow_inception_graph.pb --input_layer="input:0" --input_layer_shape="1,224,224,3" --input_layer_type="float" --output_layer="output:0"
STARTING!
Unconsumed cmdline flags: --input_layer_type=float --output_layer=output:0
Log parameter values verbosely: [0]
Graph: [../tf/tensorflow_inception_graph.pb]
Input layers: [input:0]
Input shapes: [1,224,224,3]
ERROR: The model is not a valid Flatbuffer buffer
Failed to load model ../tf/tensorflow_inception_graph.pb
Benchmarking failed```

**Comment by samveen** [comment]
After fixing the error due to missing linkage against `libatomic` for the image_label target, the build fails as below:

```
[100%] Built target tensorflow-lite
[100%] Linking CXX executable label_image
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/tools/evaluation/utils.cc.o: in function `tflite::evaluation::CreateXNNPACKDelegate(TfLiteXNNPackDelegateOptions const*)':
utils.cc:(.text+0x1b04): undefined reference to `TfLiteXnnpackDelegatePluginCApi'
collect2: error: ld returned 1 exit status
gmake[3]: *** [examples/label_image/CMakeFiles/label_image.dir/build.make:370: examples/label_image/label_image] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:6608: examples/label_image/CMakeFiles/label_image.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:6615: examples/label_image/CMakeFiles/label_image.dir/rule] Error 2
```

**Comment by pkgoogle** [comment]
I was able to replicate with linux (debian) on the nightly branch with the instructions from https://www.tensorflow.org/lite/guide/build_cmake

specifically:
```sh
git clone https://github.com/tensorflow/tensorflow.git tensorflow_src
cd tensorflow_src
git switch nightly
git pull
cd ..
mkdir tflite_build
cd tflite_build
cmake ../tensorflow_src/tensorflow/lite
cmake --build . -j
cmake --build . -j -t benchmark_model #builds properly
cmake --build . -j -t label_image #this fails
```

Error:
```sh
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/profiling/profile_summary_formatter.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/profiling/memory_info.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/tools/evaluation/utils.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/profiling/profile_summarizer.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/tools/tool_params.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/tools/delegates/xnnpack_delegate_provider.cc.o
[100%] Linking CXX executable label_image
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/tools/evaluation/utils.cc.o: in function `tflite::evaluation::CreateXNNPACKDelegate(TfLiteXNNPackDelegateOptions const*)':
utils.cc:(.text+0x2f4f): undefined reference to `TfLiteXnnpackDelegatePluginCApi'
collect2: error: ld returned 1 exit status
gmake[3]: *** [examples/label_image/CMakeFiles/label_image.dir/build.make:375: examples/label_image/label_image] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:7494: examples/label_image/CMakeFiles/label_image.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:7501: examples/label_image/CMakeFiles/label_image.dir/rule] Error 2
gmake: *** [Makefile:2522: label_image] Error 2
```

Hi @terryheo, can you please take a look? Thanks.
